corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -40,6 +42,10 @@ def process_change(self, pillow_instance, change):
         ensure_document_exists(change)
         ensure_matched_revisions(change, doc)
 
+        if is_deletion(doc):
```

Comment:
soft delete is the case that this handles.

if it's been hard deleted, `is_deletion` wouldn't handle it anyway. I think you're right that that only happens when deleting a whole domain which should have 2 other checks: (1) i believe it usually would have a month or so in the soft deleted state before we hard delete and (2) i think nickp's function searches ES as well for things to delete synchronously rather than relying on this pillow